### PR TITLE
src: Implemented es6 default parameters.

### PR DESCRIPTION
**Pull request by linbingquan**
**Description**

Followup of https://github.com/mrdoob/three.js/commit/00f2725c37e5ed1cca5edae4a9e4031ef1346683

### CODE REVIEW

File: src/animation/AnimationAction.js [hunk 1]
```diff
@@ -3,11 +3,11 @@ import { WrapAroundEnding, ZeroCurvatureEnding, ZeroSlopeEnding, LoopPingPong, L
 
 class AnimationAction {
 
-	constructor( mixer, clip, localRoot, blendMode ) {
+	constructor( mixer, clip, localRoot = null, blendMode ) {
```
Comment (mrdoob) [inline src/animation/AnimationAction.js:6]: I think this should be:

```js
constructor( mixer, clip, localRoot = null, blendMode = clip.blendMode ) {
```

File: src/cameras/PerspectiveCamera.js [hunk 2]
```diff
@@ -2,20 +2,21 @@ import { Camera } from './Camera.js';
...
+	this.near = near;
+	this.far = far;
+
 	this.zoom = 1;
 
-	this.near = near !== undefined ? near : 0.1;
-	this.far = far !== undefined ? far : 2000;
 	this.focus = 10;
 
-	this.aspect = aspect !== undefined ? aspect : 1;
```
Comment (mrdoob) [inline src/cameras/PerspectiveCamera.js:18]: Better not to change the order here.

File: src/textures/VideoTexture.js [hunk 3]
```diff
@@ -1,15 +1,10 @@
 import { RGBFormat, LinearFilter } from '../constants.js';
 import { Texture } from './Texture.js';
 
-function VideoTexture( video, mapping, wrapS, wrapT, magFilter, minFilter, format, type, anisotropy ) {
+function VideoTexture( video, mapping, wrapS, wrapT, magFilter = LinearFilter, minFilter = LinearFilter, format = RGBFormat, type, anisotropy ) {
```
Comment (Mugen87) [inline src/textures/VideoTexture.js:4]: Please avoid patters like this, see https://github.com/mrdoob/three.js/pull/20586#discussion_r514966578.

File: src/core/BufferGeometry.js [hunk 4]
```diff
@@ -101,13 +101,13 @@ BufferGeometry.prototype = Object.assign( Object.create( EventDispatcher.prototy
...
+	addGroup: function ( start, count, materialIndex = 0 ) {
 
 		this.groups.push( {
 
-			start: start,
-			count: count,
-			materialIndex: materialIndex !== undefined ? materialIndex : 0
+			start,
+			count,
+			materialIndex
```
Comment (mrdoob) [inline src/core/BufferGeometry.js:110]: Curious to see how Babel will handle this 🤓

File: src/core/Face3.js [hunk 5]
```diff
@@ -3,20 +3,20 @@ import { Vector3 } from '../math/Vector3.js';
...
+		this.materialIndex = materialIndex;
+
 		this.normal = ( normal && normal.isVector3 ) ? normal : new Vector3();
 		this.vertexNormals = Array.isArray( normal ) ? normal : [];
 
 		this.color = ( color && color.isColor ) ? color : new Color();
 		this.vertexColors = Array.isArray( color ) ? color : [];
 
-		this.materialIndex = materialIndex !== undefined ? materialIndex : 0;
-
```
Comment (mrdoob) [inline src/core/Face3.js:19]: Better not to change the order here either.

File: src/extras/core/CurvePath.js [hunk 6]
```diff
@@ -154,9 +154,9 @@ CurvePath.prototype = Object.assign( Object.create( Curve.prototype ), {
 
 	},
 
-	getPoints: function ( divisions ) {
+	getPoints: function ( divisions = 12 ) {
 
-		divisions = divisions || 12;
+		divisions = divisions;
```
Comment (mrdoob) [inline src/extras/core/CurvePath.js:159]: This line can be deleted.

File: src/textures/Texture.js [hunk 7]
```diff
@@ -17,30 +17,33 @@ import { ImageUtils } from '../extras/ImageUtils.js';
...
-	this.wrapS = wrapS !== undefined ? wrapS : ClampToEdgeWrapping;
-	this.wrapT = wrapT !== undefined ? wrapT : ClampToEdgeWrapping;
-
-	this.magFilter = magFilter !== undefined ? magFilter : LinearFilter;
-	this.minFilter = minFilter !== undefined ? minFilter : LinearMipmapLinearFilter;
+	// Values of encoding !== THREE.LinearEncoding only supported on map, envMap and emissiveMap.
+	//
+	// Also changing the encoding after already used by a Material will not automatically make the Material
+	// update. You need to explicitly call Material.needsUpdate to trigger it to recompile.
+	this.encoding = encoding;
```
Comment (mrdoob) [inline src/textures/Texture.js:42]: Please, do not change the order of the code.

**Comment by mrdoob** [comment]
Way too many blocking changes in this PR...
It would have been better to do different PRs instead.

**Comment by linbingquan** [comment]
> Way too many blocking changes in this PR...
> It would have been better to do different PRs instead.

OK

File: src/cameras/PerspectiveCamera.js (hunk 2 above)
Comment (linbingquan) [reply src/cameras/PerspectiveCamera.js:18]: Yes, It will be easily to review.

File: src/core/BufferGeometry.js (hunk 4 above)
Comment (gkjohnson) [reply src/core/BufferGeometry.js:110]: Babel should handle this just fine, no? It's valid JS.

Comment (mrdoob) [reply src/core/BufferGeometry.js:110]: Yep!

**Pull request closed without merging**